src/adapters/http/server.ts

```diff
@@ -30,14 +31,75 @@ class HttpAdapter extends Adapter {
     const optionsPort = httpOptions?.port
     const port = optionsPort || asyncapiServerPort
 
-    httpServer.on('request', (req, res) => {
+    httpServer.on('request', async (req, res) => {
       res.setHeader('Content-Type', 'application/json')
+
       const bodyBuffer = []
       let body: object
       req.on('data', (chunk) => {
         bodyBuffer.push(chunk)
       })
-      req.on('end', () => {
+
+      function done() {
```

Comment:
please extract the authentication logic in another class. I am thinking something like 🤔 :
```ts
class Authenticator {
    AsyncAPIServer: any; 
    parsedAsyncAPI: any;
    serverName: string;
    constructor({...})

    authenticate(req: any): Promise<void>;
    checkAuthPresence(): boolean;
}

```